src/lpython/parser/tokenizer.re

```diff
@@ -262,6 +262,9 @@ int Tokenizer::lex(Allocator &al, YYSTYPE &yylval, Location &loc, diag::Diagnost
             }
             end { RET(END_OF_FILE); }
             whitespace {
+                if(cur[0] == '#') {
+                    continue;
+                }
```

Comment:
Previously, the following was recognized as `TK_EOLCOMMENT`
```python
     # Comment
```
After making the above change, it is recognized as `TK_COMMENT`
This change was made to recognize the comment after the colon 
```python 
for i in range(N): # type: parallel
```